--- a/printer_alerts      2016-04-01 16:35:40.000000000 +0200
+++ b/printer_alerts      2016-06-08 10:37:50.000000000 +0200
@@ -236,6 +236,7 @@
 check_info["printer_alerts"] = {
     'check_function':          check_printer_alerts,
     'inventory_function':      inventory_printer_alerts,
+    'handle_empty_info':       True,
     'service_description':     'Alerts',
     'snmp_info':               ('.1.3.6.1.2.1.43.18.1.1', [
                                    '2', # prtAlertSeverityLevel
