--- a/win_netstat    2016-04-01 16:35:40.000000000 +0200
+++ b/win_netstat    2016-06-08 12:53:08.000000000 +0200
@@ -54,9 +54,9 @@

 win_netstat_states = {
   # German
-  "ABH\x99REN"  : "LISTENING",
-  "HERGESTELLT" : "ESTABLISHED",
-  "WARTEND"     : "TIME_WAIT",
+  u"ABH\x99REN"  : "LISTENING",
+  u"HERGESTELLT" : "ESTABLISHED",
+  u"WARTEND"     : "TIME_WAIT",
   # Add further states in any required language here. Sorry, Windows
   # has no "unset LANG" ;-)
 }
