--- ~/share/check_mk/checks/ps.include	2015-07-16 14:12:33.000000000 +0200
+++ ~/local/share/check_mk/checks/ps.include	2015-07-26 11:10:55.303386400 +0200
@@ -206,6 +206,12 @@
 #   "user" : "foo",
 #   "levels" : (1, 1, 99999, 99999)
 # }
+
+# Params Extension for single Process memory and cpu
+# {
+#   "singleproc" : True,
+# }
+
 def parse_ps_time(text):
     if "-" in text:
         tokens = text.split("-")
@@ -250,6 +256,9 @@
     percent_cpu    = 0.0
     extended_perfdata = False
 
+    if "singleproc" not in params:
+        params["singleproc"] = False
+
     # The counter names for the ps check are quite volatile, because there is
     # dynamic part (the pid) in the name. Therefore we clear any counters
     # older than one day. Affected are ps_wmic.user, ps_wmic.kernel, ps_stat.pcpu
@@ -266,8 +275,14 @@
             if ps[0][0] == '(':
                 addinfo = ps[0][1:-1].split(",")
                 if len(addinfo) >= 4: # extended performance data: virtualsize, residentsize, %cpu
-                    virtual_size += int(addinfo[1])  # kB
-                    resident_size += int(addinfo[2]) # kB
+                    if params.get("singleproc") != True:
+                        virtual_size += int(addinfo[1])  # kB
+                        resident_size += int(addinfo[2]) # kB
+                    else:
+                        if virtual_size <= int(addinfo[1]):
+                            virtual_size = int(addinfo[1])
+                        if resident_size <= int(addinfo[2]):
+                            resident_size = int(addinfo[2])
                     if len(addinfo) >= 10: # even more data: processId, pagefile_usage, usermodetime, kernelmodetime, threadCount, openHandles
                         pid, pagefile_usage, user_c, kernel_c, handle_c = map(int, addinfo[4:9])
                         counter_wrapped = False
@@ -287,8 +302,14 @@
 
                         user_perc = user_per_sec / 100000.0 / cpu_cores
                         kernel_perc = kernel_per_sec / 100000.0 / cpu_cores
-                        percent_cpu += user_perc + kernel_perc
-                        handle_count += handle_c
+                        if params.get("singleproc") != True:
+                            percent_cpu += user_perc + kernel_perc
+                            handle_count += handle_c
+                        else:
+                            if percent_cpu <= user_perc + kernel_perc:
+                                percent_cpu = user_perc + kernel_perc
+                            if handle_count <= handle_c:
+                                handle_count = handle_c
                     else:
                         if ":" in addinfo[3]:
                             if '/' in addinfo[3]:
@@ -344,6 +365,8 @@
         infotext += " (ok from %d to %d)(!)" % (okmin, okmax)
 
     if virtual_size:
+        if params.get("singleproc") == True:
+            infotext += ", maximum single process"
         for size, name in [ (virtual_size, "virtual"), (resident_size, "resident") ]:
             infotext += " %.1f MB %s" % ((size / 1024.0), name)
             if "%s_levels" % name in params:
