From 3428cfdf653f8f1ea3fc3270dd777e86daa2433b Mon Sep 17 00:00:00 2001
From: Andreas Doehler <andreas.doehler@gmail.com>
Date: Tue, 25 Mar 2014 14:44:42 +0100
Subject: [PATCH 2/2] Plugin can submit code page information for plugin output

Signed-off-by: Andreas Doehler <andreas.doehler@gmail.com>
---
 modules/check_mk_base.py | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/modules/check_mk_base.py b/modules/check_mk_base.py
index d780d9e..d7cb5df 100644
--- a/modules/check_mk_base.py
+++ b/modules/check_mk_base.py
@@ -746,6 +746,7 @@ def parse_info(lines, hostname):
     section = []
     section_options = {}
     separator = None
+    codepage = None
     for line in lines:
         if line[:4] == '<<<<' and line[-4:] == '>>>>':
             host = line[4:-4]
@@ -787,8 +788,17 @@ def parse_info(lines, hostname):
             if "persist" in section_options:
                 until = int(section_options["persist"])
                 persist[section_name] = ( until, section )
+            
+            try:
+                codepage = int(section_options["code"])
+            except:
+                codepage = None
 
         elif line != '':
+            if codepage != None:
+                cp = 'cp%s' % codepage
+                line = line.decode(cp)
+                line = line.encode('utf-8')
             section.append(line.split(separator))
     return info, piggybacked, persist
 
-- 
1.8.5.2

